adding reserved keywords to docs

elubow opened the pull request:
This deals with thinkaurelius/titan#1160 by adding a reserved keywords section to the docs.  Happy to cleanup further if you don't like the language.

elubow committed:
adding reserved keywords to docs

titan-cla commented:
Hi @elubow, thanks for your contribution!

In order for us to evaluate and accept your PR, we ask that you [sign a contribution license agreement](http://titan-cla.datastax.com). It's all electronic and will take just minutes.

titan-cla commented:
You did it @elubow!

Thank you for signing the Contribution License Agreement.

dalaro commented:
Could you please make the following changes?
- Emphasize that the list of names is reserved for all schemata: vertex labels, edge labels, and property keys
- typo: regareding
- To sync with the rest of the docs, leave "Vertex" not capitalized unless it starts a sentence -- "vertex labels" is fine in the middle of a sentence

elubow commented:
Good catch on the typo. Added the rest of the schemata.  Also moved the example to make it more clear by using the word `property` with a `vertex`.  Squashed the change in to the original commit for clarity.
